### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by wadaudzai**
In this part of the project me and my dear teammate @Peter1907 created unit tests for all the methods in the project.
Dear reviewer, we have worked on this project to the best of our knowledge please check it if you got time. Thanks for your time.

Regards

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    it 'returns correct names of person object instance variables' do
+      expect(person.age).to eq 20
+      expect(person.name).to eq 'John'
+    end
+
+    it 'should return true if the age is greater than or equal to 18' do
+      expect(person.can_use_services?).to eq(true)
+    end
+  end
```
Comment (youmari) [inline spec/person_spec.rb]: - [x] as required you should test all the methods in this class you forgot some of them so please make sure to write unit tests for them as well in order to match the requirements 
- missing tests for: 
    - `of_age?` `add_rental` `correct_name`
    
Screenshot 2022-11-24 at 17 56 16

File: spec/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+describe 'Book' do
+  context 'should create an instance of Book class' do
+    book = Book.new('Hobbit', 'Jhon')
+
+    it 'takes two parameters and returns a Book object' do
+      expect(book.class) == Book
+    end
+  end
+end
```
Comment (youmari) [inline spec/book_spec.rb]: - [x] as required you should test all the methods in this class you forgot some of them so please make sure to write unit tests for them as well in order to match the requirements 
- missing tests for: 
    -  `add_rental`
    
Screenshot 2022-11-24 at 17 56 16

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @wadaudzai @Peter1907 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/book_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      expect(book.class) == Book
+    end
+    it 'returns correct names of book object instance variables' do
+      expect(book.title).to eq 'Hobbit'
+      expect(book.author).to eq 'Jhon'
+    end
+    it 'should return true if the age is greater than or equal to 18' do
+      expect(book.rentals).to eq([])
+    end
+  end
```
Comment (thecodechaser) [inline spec/book_spec.rb]: - [ ] Please change the title for this test, It should be related and meaningful for the test. Right now it's related to the age test.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @wadaudzai, and @Peter1907👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- All tests are passed. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ ] [optional] I would suggest using this new Microverse README [template ](https://github.com/microverseinc/readme-template), It will make the README file looks more professional.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @wadaudzai 

Your project is complete! Great job for making the requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- test ok
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

- [optional] please kindly update your readme with the recent [template](https://github.com/microverseinc/readme-template).

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**